pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,996 @@
+        saved to `fp[fp.samples_group/{param}/temp{k}].attrs['acl']`; the
+        maximum over all of the temperatures and walkers is saved to
+        `fp[fp.samples_group/{param}].attrs['acl']`; the maximum over all the
+        parameters, temperatures, and walkers is saved to the file's 'acl'
+        attribute.
+
+        Parameters
+        ----------
+        fp : InferenceFile
+            An open file handler to write the samples to.
+        acls : WaveformArray
+            An array of autocorrelation lengths (the sort of thing returned by
+            `compute_acls`).
+
+        Returns
+        -------
+        acl
+            The maximum of the acls that was written to the file.
+        """
+        # write the individual acls
+        pgroup = fp.samples_group + '/{param}'
+        tgroup = pgroup + '/temp{tk}'
+        group = tgroup + '/walker{wi}'
+        tidx = numpy.arange(fp.ntemps)
+        overall_max = 0
+        for param in acls.fieldnames:
+            max_acls = []
+            for tk in tidx:
+                max_acl = 0
+                for wi,acl in enumerate(acls[param][tk,:]): 
+                    fp[group.format(param=param, tk=tk,
+                                    wi=wi)].attrs['acl'] = acl
+                    max_acl = max(max_acl, acl)
+                # write the maximum over the walkers
+                fp[tgroup.format(param=param, tk=tk)].attrs['acl'] = max_acl
+                max_acls.append(max_acl)
+            # write the maximum over the temperatures
+            this_max = max(max_acls)
+            fp[pgroup.format(param=param)].attrs['acl'] = this_max
+            overall_max = max(overall_max, this_max)
+            # write the maximum over the params
+            fp[pgroup.format(param=param)].attrs['acl'] = max_acl
+        # write the maximum over all params
+        fp.attrs['acl'] = overall_max
+        return fp.attrs['acl']
+
+    @staticmethod
+    def read_acls(fp):
+        """Reads the acls of all the walker chains saved in the given file.
+
+        Parameters
+        ----------
+        fp : InferenceFile
+            An open file handler to read the acls from.
+
+        Returns
+        -------
+        WaveformArray
+            An ntemps x nwalkers `WaveformArray` containing the acls for
+            every temp and walker, with the variable arguments as fields.
+        """
+        group = fp.samples_group + '/{param}/temp{tk}/walker{wi}'
+        widx = numpy.arange(fp.nwalkers)
+        tidx = numpy.arange(fp.ntemps)
+        arrays = {}
+        for param in fp.variable_args:
+            arrays[param] = numpy.array([
+                [fp[group.format(param=param, tk=tk, wi=wi)].attrs['acl']
+                    for wi in widx]
+                 for tk in tidx])
+        return WaveformArray.from_kwargs(**arrays)
+
+    @classmethod
+    def calculate_logevidence(cls, fp, thin_start=None, thin_end=None,
+            thin_interval=None):
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->